### ISSUE TITLE
Bootstrapped core dependencies: context and nats connection.

### ISSUE THREAD

**Issue by n-holmstedt** [shono-io/nats-ai#1]
Right now there isn't any global context created while bootstrapping the application, which causes any extension needing a context to create its own. 

Also, any extensions using NATS would want to use the same connection object, so this would probably be well suited to create somewhere in the initiation process as well so it could be passed down as a dependency, much like the context issue above.

**Comment by calmera** [comment]
Yeah, I did hack this version together, but will need to structure it a bit better.
